uiowa/uiowa
# Aggregator Extended

Resolves #7068
Resolves #5615

# To Test

```
ddev composer install && ddev blt ds --site=sandbox.uiowa.edu && ddev drush @sandbox.local uli aggregator/sources/31/configure
```

See the purge items checkbox and check it. Change the interval to 15 min. Save the source.
Go to https://sandbox.uiowa.ddev.site/admin/config/services/aggregator and note the number of items in the campuslabs table. Notice the number in a few other tables (e.g. HR liveWell, USG events, etc.)

`ddev drush @sandbox.local cr && ddev drush @sandbox.local cron` _(ignore the htaccess error)_

Note the dramatic reduction in campuslabs items.

Manually "refresh items" should also work.

To test the upstream feed better, you can use https://sandbox.prod.drupal.uiowa.edu and add/remove article content. Keep in mind the current 1 year limit. Even though it might be in the feed, it is no longer imported given how old it is. (Sandbox only had 4 items when I tested)

# Deep Dive

With xdebug on, and breakpoints set at docroot/modules/contrib/aggregator/src/ItemsImporter.phpL105 and docroot/modules/custom/uiowa_aggregator/src/ItemsImporterOverride.phpL45 you will notice that on cron run or the update items operation it steps into ItemsImporterOverride refresh first and then proceeds to do the rest of the refresh method in ItemsImporter meaning there is no functionality lost by hijacking with the service decorator.
